src/IO/CachedInMemoryReadBufferFromFile.cpp

```diff
@@ -26,6 +29,19 @@ String CachedInMemoryReadBufferFromFile::getFileName() const
     return in->getFileName();
 }
 
+String CachedInMemoryReadBufferFromFile::getInfoForLog()
+{
+    return "CachedInMemoryReadBufferFromFile(" + in->getInfoForLog() + ")";
+}
+
+bool CachedInMemoryReadBufferFromFile::isSeekCheap()
+{
+    /// This is a weird method. Seek to what offset? If the target offset is in cache then the seek
+    /// is cheap. Let's assume that the caller means a short seek that would stay within
+    /// working_buffer if it's not empty.
+    return available() == 0;
```

Comment:
you mean `available() != 0` ?
why not just `buffer().size() != 0`?

This only affects increase of some profile events and in one case it can force the usage of lazy seek, instead of seeking we ignore certain amount of bytes until we get to the offset.
Lazy seek is also limited by `remote_read_min_bytes_for_seek`

Reply:
Oops, yes, comment didn't match the code. Changed to a better heuristic.